plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -86,9 +86,9 @@ fu_plugin_synaptics_add_device (FuPlugin *plugin,
 	board_str = synapticsmst_device_board_id_to_string (synapticsmst_device_get_board_id (device));
 	name = g_strdup_printf ("Synaptics %s inside %s", synapticsmst_device_get_chip_id (device),
 				board_str);
-	guid_str =  synapticsmst_device_get_guid (device);
-	if (guid_str == NULL) {
-		g_debug ("invalid GUID for board ID %x",
+	guids = synapticsmst_device_get_guids (device);
+	if (guids == NULL) {
```

Comment:
Don't you want to check for `guids->len == 0` -- `guids` should always be valid as it's created in _init.